src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -221,7 +240,11 @@ private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModul
                 moduleBeingBuilt.SetMethodBody(synthesizedEntryPoint, emittedBody);
             }
 
-            Debug.Assert((object)entryPoint != null || entryPointAndDiagnostics.Diagnostics.HasAnyErrors() || !compilation.Options.Errors.IsDefaultOrEmpty);
+            if (addedDefinition && moduleBeingBuilt != null)
+            {
+                moduleBeingBuilt.AddSynthesizedDefinition(entryPoint.ContainingType, synthesizedEntryPoint);
+            }
```

Comment:
Could `addedDefinition` be removed and this `if` moved to where `addedDefinition` is set?